packages/appium/lib/extension/driver-config.js

```
@@ -0,0 +1,215 @@
+   */
+  getConfigProblems (extData) {
+    const problems = [];
+    const {platformNames, automationName} = extData;
+
+    if (!_.isArray(platformNames)) {
+      problems.push({
+        err: 'Missing or incorrect supported platformNames list.',
+        val: platformNames
+      });
+    } else {
+      if (_.isEmpty(platformNames)) {
+        problems.push({
+          err: 'Empty platformNames list.',
+          val: platformNames
+        });
+      } else {
+        for (const pName of platformNames) {
+          if (!_.isString(pName)) {
+            problems.push({err: 'Incorrectly formatted platformName.', val: pName});
+          }
+        }
+      }
+    }
+
+    if (!_.isString(automationName)) {
+      problems.push({err: 'Missing or incorrect automationName', val: automationName});
+    }
+
+    if (this.knownAutomationNames.has(automationName)) {
+      problems.push({
+        err: 'Multiple drivers claim support for the same automationName',
+        val: automationName
+      });
+    }
+
+    // should we retain the name at the end of this function, once we've checked there are no problems?
+    this.knownAutomationNames.add(automationName);
+
+    return problems;
+  }
+
+  /**
+   * @param {ExtName<DriverType>} driverName
+   * @param {ManifestDriverData} extData
+   * @returns {string}
+   */
+  extensionDesc (driverName, {version, automationName}) {
+    return `${driverName}@${version} (automationName '${automationName}')`;
+  }
+
+  /**
+   * Given capabilities, find a matching driver within the config. Load its class and return it along with version and driver name.
+   * @param { { automationName: string, platformName: string } } caps
+   * @returns { { driver: import('./manifest').DriverClass, version: string, driverName: string } }
+   */
+  findMatchingDriver ({automationName, platformName}) {
+    if (!_.isString(platformName)) {
+      throw new Error('You must include a platformName capability');
```

not sure if we need to throw a generic error here or more specific one from the list of available W3C error codes

---

old code